### PR TITLE
Create `Image` component

### PR DESCRIPTION
**Pull request by manuelpuyol**
Creates a basic `Image` component to render `` tags.
The component can be useful to enforce rules, like always having an `alt` attribute.

I also provide the `primer_image` helper for easier use.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
     *Kristján Oddsson*
 
+* Create `Image` component.
+
```
Comment (koddsson) [inline CHANGELOG.md:12]: Missing your name?

File: app/components/primer/image.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    #   <%= render(Primer::Image.new(src: "https://github.com/github.png", alt: "GitHub", height: 100, width: 100)) %>
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(src:, alt:, loading: DEFAULT_LOADING, **system_arguments)
+      @system_arguments = system_arguments
+
+      @system_arguments[:tag] = :img
+      @system_arguments[:src] = src
+      @system_arguments[:alt] = alt
+      @system_arguments[:loading] = fetch_or_fallback(LOADING_OPTIONS, loading, DEFAULT_LOADING)
```
Comment (koddsson) [inline app/components/primer/image.rb:35]: Could this be something like: 

```suggestion
      @system_arguments[:loading] = fetch_or_fallback(LOADING_OPTIONS, loading, DEFAULT_LOADING) if loading != :eager
```

Since eager loading is the default, being less verbose will reduce confusion when reading the compiled HTML.

File: component_generator.thor [hunk 3]
```diff
@@ -42,9 +42,9 @@ class ComponentGenerator < Thor::Group
     template("templates/stories.tt", "stories/primer/#{underscore_name}_stories.rb")
   end
 
-  def add_to_rakefile
-    insert_into_file("Rakefile", "      Primer::#{class_name},\n", after: "    components = [\n")
-    insert_into_file("Rakefile", "      Primer::#{class_name},\n", after: "js_components = [\n", force: true) if js_package_name
+  def add_to_docs_rakefile
+    insert_into_file("lib/tasks/docs.rake", "      Primer::#{class_name},\n", after: "    components = [\n")
+    insert_into_file("lib/tasks/docs.rake", "      Primer::#{class_name},\n", after: "js_components = [\n", force: true) if js_package_name
```
Comment (koddsson) [inline component_generator.thor:47]: These changes seem unrelated to the rest of the pull request. What do you think about putting it in its own pull request?

File: stories/primer/image_stories.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      classes "custom-class"
+      src "https://github.com/github.png"
+      alt "image description"
+      height 100
+      width 100
+      select(:loading, Primer::Image::LOADING_OPTIONS, Primer::Image::DEFAULT_LOADING)
+    end
+
+    content do
+      "Update your stories!"
```
Comment (koddsson) [inline stories/primer/image_stories.rb:17]: Does this need updating?

File: stories/primer/image_stories.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Primer::ImageStories < ViewComponent::Storybook::Stories
+  layout "storybook_preview"
+
+  story(:image) do
+    controls do
+      classes "custom-class"
+      src "https://github.com/github.png"
+      alt "image description"
```
Comment (koddsson) [inline stories/primer/image_stories.rb:10]:
```suggestion
      alt "The GitHub logo"
```

File: stories/primer/image_stories.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Primer::ImageStories < ViewComponent::Storybook::Stories
+  layout "storybook_preview"
+
+  story(:image) do
+    controls do
+      classes "custom-class"
```
Comment (koddsson) [inline stories/primer/image_stories.rb:8]:
```suggestion
      classes ""
```

Or even remove?

File: templates/test.tt [hunk 7]
```diff
@@ -6,6 +6,8 @@ class Primer<%= class_name %>Test < Minitest::Test
   include Primer::ComponentTestHelpers
 
   def test_renders
+    render_inline(Primer::<%= class_name %>.new)
```
Comment (koddsson) [inline templates/test.tt:9]: This change also seems unrelated to the rest of the pull request.

File: test/components/component_test.rb [hunk 8]
```diff
@@ -7,6 +7,7 @@ class PrimerComponentTest < Minitest::Test
 
   # Components with any arguments necessary to make them render
   COMPONENTS_WITH_ARGS = [
+    [Primer::Image, {  src: "src", alt: "alt" }],
```
Comment (koddsson) [inline test/components/component_test.rb:10]:
```suggestion
    [Primer::Image, { src: "src", alt: "alt" }],
```

File: app/components/primer/image.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    #
+    # @example Lazy loading
+    #
+    #   <%= render(Primer::Image.new(src: "https://github.com/github.png", alt: "GitHub", loading: :lazy)) %>
+    #
+    # @example Custom size
+    #
+    #   <%= render(Primer::Image.new(src: "https://github.com/github.png", alt: "GitHub", height: 100, width: 100)) %>
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
```
Comment (koddsson) [inline app/components/primer/image.rb:28]: `src`, `alt` and `loading` are missing.

File: app/components/primer/image.rb [hunk 10]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use `Image` to render images. It can be rendered using the `primer_image` helper.
+  #
+  # @accessibility
+  #   Always provide a meaningful `alt` for your images.
```
Comment (joelhawksley) [inline app/components/primer/image.rb:7]:
```suggestion
  #   Always provide a meaningful `alt`.
```

File: app/components/primer/image.rb [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    def initialize(src:, alt:, loading: DEFAULT_LOADING, **system_arguments)
+      @system_arguments = system_arguments
+      @loading = fetch_or_fallback(LOADING_OPTIONS, loading, DEFAULT_LOADING)
+
+      @system_arguments[:tag] = :img
+      @system_arguments[:src] = src
+      @system_arguments[:alt] = alt
+
+
+      @system_arguments[:loading] = @loading if @loading == :lazy
```
Comment (joelhawksley) [inline app/components/primer/image.rb:41]: Do we even need the `:eager` option any more? Could we instead have a `lazy` argument to the initializer?

**Review by joelhawksley — COMMENTED** [review]
Looking good! 

cc @tobiasahlin as this is related to your recent work on images. Perhaps you might have some ideas for Manuel here?

**Comment by tobiasahlin** [comment]
👀 @manuelpuyol maybe we could consider setting [`decoding="async"`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decoding) here whenever we're setting `loading="lazy"`, to further reduce delay in presenting other content (especially for browsers that don't support `loading="lazy"` yet)

File: app/components/primer/image.rb [hunk 12]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use `Image` to render images. It can be rendered using the `primer_image` helper.
```
Comment (joelhawksley) [inline app/components/primer/image.rb:4]: I wonder if we should make this simpler and instead come up with a standard way of indicating the helper method name somewhere else?

```suggestion
  # Use `Image` to render images.
```

**Review by joelhawksley — APPROVED** [review]
